compiler/rustc_ast/src/ast.rs

```diff
@@ -3462,7 +3462,7 @@ pub struct FnHeader {
     /// Whether this is `async`, `gen`, or nothing.
     pub coroutine_kind: Option<CoroutineKind>,
     /// The `const` keyword, if any
-    pub constness: Const,
+    pub constness: BoundConstness,
```

Comment:
This is not a correct usage of `BoundConstness`, as it's not a bound, but adding another enum that has exactly the same variants seemed kinda bad, too. So I'm reusing it for now, but I can also rename it or duplicate it in a follow-up commit in this PR